relay-event-normalization/src/normalize/span/description/sql/parser.rs

```diff
@@ -116,21 +140,23 @@ impl NormalizeVisitor {
         Self::collapse_items(&mut collapse, &mut select.projection);
     }
 
-    fn transform_from(from: &mut [TableWithJoins]) {
+    fn transform_from(&mut self, from: &mut [TableWithJoins]) {
         // Iterate "FROM".
         for from in from {
-            Self::transform_table_factor(&mut from.relation);
+            self.transform_table_factor(&mut from.relation);
             for join in &mut from.joins {
-                Self::transform_table_factor(&mut join.relation);
+                self.transform_table_factor(&mut join.relation);
             }
         }
     }
 
-    fn transform_table_factor(table_factor: &mut TableFactor) {
+    fn transform_table_factor(&mut self, table_factor: &mut TableFactor) {
         match table_factor {
             // Recurse into subqueries.
             TableFactor::Derived { subquery, .. } => {
-                Self::transform_query(subquery);
+                self.current_depth += 1;
+                self.transform_query(subquery);
+                self.record_depth();
```

Comment:
We update the depth on pre/post expression visiting, is repeating it here redundant?

Reply:
I was trying to measure the level of nesting of not just expressions, but also subqueries. I reverted that part now though because it made the measured depth kind of voodoo.